src/Faithlife.Tracing.AspNetCore/AspNetCoreTracing.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Globalization;
+using System.Threading;
+using Faithlife.Tracing.AspNet;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Faithlife.Tracing.AspNetCore
+{
+	public static class AspNetCoreTracing
+	{
+		public static void UseTracing(this IApplicationBuilder app, Action<AspNetCoreTracingSettings> configure)
+		{
+			var settings = new AspNetCoreTracingSettings();
+			configure(settings);
+
+			var serviceName = settings.ServiceName ?? throw new ArgumentOutOfRangeException(nameof(settings.ServiceName));
+
+			var actualSamplingRate = settings.SamplingRate ?? 1;
+			SamplingRate =
+				actualSamplingRate < 0 ? throw new ArgumentOutOfRangeException(nameof(settings.SamplingRate), "SamplingRate cannot be negative") :
+				actualSamplingRate > 1 ? throw new ArgumentOutOfRangeException(nameof(settings.SamplingRate), "SamplingRate must be between 0.0 and 1.0") :
+				(int) Math.Round(actualSamplingRate * c_samplingPrecision);
+			Tracer = settings.CreateTracer?.Invoke();
+
+			app.Use(async (httpContext, next) =>
+			{
+				var headers = httpContext.Request.Headers;
+				var parentTrace = Tracer.ExtractTrace(x => headers[x]);
+				if (parentTrace != null || headers["X-B3-Flags"] == "1" || GetHashCode(Interlocked.Increment(ref s_requestCount)) % c_samplingPrecision < SamplingRate)
```

Comment:
Should the sampling code be extracted to a shared `Sampler` class?

Reply:
Probably. Doesn't need to be done in this PR, though.